File: app/src/main/java/com/digitex/designertools/ui/ScreenRecordRequestActivity.java
package com.digitex.designertools.ui;

import android.app.Activity;
import android.content.Intent;
import android.media.projection.MediaProjectionManager;
import android.os.Bundle;

import com.digitex.designertools.DesignerToolsApplication;
import com.digitex.designertools.overlays.ColorPickerOverlay;
import com.digitex.designertools.utils.PreferenceUtils.ColorPickerPreferences;

public class ScreenRecordRequestActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MediaProjectionManager mpm = getSystemService(MediaProjectionManager.class);
        startActivityForResult(mpm.createScreenCaptureIntent(), 42);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (resultCode == RESULT_OK) {
            ((DesignerToolsApplication) getApplication()).setScreenRecordPermissionData(
                    resultCode, data);
            Intent newIntent = new Intent(this, ColorPickerOverlay.class);
            this.startService(newIntent);
            ColorPickerPreferences.setColorPickerActive(this, true);
        }
        finish();
    }
}

File: app/src/main/java/com/digitex/designertools/ui/ScreenRecordRequestActivity.kt
package com.digitex.designertools.ui

import android.content.Intent
import android.media.projection.MediaProjectionManager
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.getSystemService

import com.digitex.designertools.designerApplication
import com.digitex.designertools.overlays.ColorPickerOverlay
import com.digitex.designertools.utils.PreferenceUtils.ColorPickerPreferences

class ScreenRecordRequestActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val mpm = getSystemService<MediaProjectionManager>()!!
        startActivityForResult(mpm.createScreenCaptureIntent(), 42)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (resultCode == RESULT_OK) {
            designerApplication.setScreenRecordPermissionData(resultCode, data)
            startService(Intent(this, ColorPickerOverlay::class.java))
            ColorPickerPreferences.setColorPickerActive(this, true)
        }
        super.onActivityResult(requestCode, resultCode, data)
        finish()
    }
}
